railties/lib/rails/generators/rails/app/templates/config/puma.rb.tt

```diff
@@ -11,6 +11,12 @@ max_threads_count = ENV.fetch("RAILS_MAX_THREADS") { 5 }
 min_threads_count = ENV.fetch("RAILS_MIN_THREADS") { max_threads_count }
 threads min_threads_count, max_threads_count
 
+# Specifies that the worker count should equal the number of processors in production.
+if ENV["RAILS_ENV"] == "production"
+  worker_count = ENV.fetch("WEB_CONCURRENCY") { Concurrent.physical_processor_count }
+  workers worker_count if worker_count > 1
```

Comment:
I suspect this does not work, because `ENV.fetch("WEB_CONCURRENCY")` returns a string if the variable is set, which you cannot compare with an integer:

```
~ % WEB_CONCURRENCY=1 ruby -e 'p ENV.fetch("WEB_CONCURRENCY")'
"1"
```

Reply:
😬

Reply:
Good point, will fix.